src/logic/actions/ViewPortActions.ts

```diff
@@ -0,0 +1,159 @@
+import {EditorModel} from "../../staticModels/EditorModel";
+import {NumberUtil} from "../../utils/NumberUtil";
+import {ViewPointSettings} from "../../settings/ViewPointSettings";
+import {ISize} from "../../interfaces/ISize";
+import {IRect} from "../../interfaces/IRect";
+import {ImageUtil} from "../../utils/ImageUtil";
+import {RectUtil} from "../../utils/RectUtil";
+import {IPoint} from "../../interfaces/IPoint";
+import {PointUtil} from "../../utils/PointUtil";
+import {SizeUtil} from "../../utils/SizeUtil";
+import {EditorActions} from "./EditorActions";
+import {Direction} from "../../data/enums/Direction";
+import {DirectionUtil} from "../../utils/DirectionUtil";
+
+export class ViewPortActions {
+    public static updateViewPortSize() {
+        if (!!EditorModel.editor) {
+            EditorModel.viewPortSize = {
+                width: EditorModel.editor.offsetWidth,
+                height: EditorModel.editor.offsetHeight
+            }
+        } else {
+            return null;
+        }
+    }
+
+    public static updateDefaultViewPortImageRect() {
+        if (!!EditorModel.viewPortSize && !!EditorModel.image) {
+            const minMargin: IPoint = {x: ViewPointSettings.CANVAS_MIN_MARGIN_PX, y: ViewPointSettings.CANVAS_MIN_MARGIN_PX};
+            const realImageRect: IRect = {x: 0, y: 0, ...ImageUtil.getSize(EditorModel.image)};
+            const viewPortWithMarginRect: IRect = {x: 0, y: 0, ...EditorModel.viewPortSize};
+            const viewPortWithoutMarginRect: IRect = RectUtil.expand(viewPortWithMarginRect, PointUtil.multiply(minMargin, -1));
+            EditorModel.defaultRenderImageRect = RectUtil.fitInsideRectWithRatio(viewPortWithoutMarginRect, RectUtil.getRatio(realImageRect));
+        } else {
+            return null;
+        }
+    }
+
+    public static calculateViewPortContentSize(): ISize {
+        if (!!EditorModel.viewPortSize && !!EditorModel.image) {
+            const defaultViewPortImageRect: IRect = EditorModel.defaultRenderImageRect;
+            const scaledImageSize: ISize = SizeUtil.scale(EditorModel.defaultRenderImageRect, EditorModel.zoom);
+            return {
+                width: scaledImageSize.width + 2 * defaultViewPortImageRect.x,
```

Comment:
czym jest ta "2"? xD

Reply:
ta dwójka to prosze pana jest związana z marginesem to znaczy jak obliczam rozmiar całości to margin wcodzi dwa razy

Reply:
oki